### PR TITLE
Improve precompiles typings by utilizing enums consistently

### PR DESCRIPTION
**Pull request by Svetomech**
## Summary by CodeRabbit

- **New Features**
	- Introduced several new events: `ActionStateChange`, `ActionVoted`, `CreateAction`, `CreateTemplate`, and `UpdateTemplate`.
	- Enhanced ABI definitions for functions related to actions and templates.
	- Added a new feature allowing contracts to approve actions.

- **Bug Fixes**
	- Improved type safety by changing data types from `int32` to `uint8` for various fields across multiple structures and events.

- **Documentation**
	- Updated comments and documentation to reflect the new data types and structures.

- **Tests**
	- Adjusted test cases to align with the new data types, ensuring consistency and correctness in assertions.

### CODE REVIEW

File: precompiles/justfile [hunk 1]
```diff
@@ -4,15 +4,26 @@ tmpdir := `mktemp -d`
 all: abi
 
 _tools:
+  # Install solc if not present
+  command -v solc || brew install solidity
```
[the remark this answers is not in the document]
Comment (pitasi) [reply precompiles/justfile]: as the bot suggested, this only works if using `brew`, which is not always the case, so i'd refrain that. Maybe just print an error message that `solc` is missing and should be installed, with a suggestion to run `brew install solidity` on mac os

**Review by mn13 — CHANGES_REQUESTED** [review]
Can you please add change log?

By experience with last release, it should be under "Consensus Breaking Changes"

**Review by pitasi — CHANGES_REQUESTED** [review]

File: tests/cases/warden_precompile_action.go [hunk 2]
```diff
@@ -175,15 +175,15 @@ func (c *Test_WardenPrecompileAction) Run(t *testing.T, ctx context.Context, bui
...
+		require.Equal(t, newKeyRequestEvents[0].KeyType, uint8(types.KeyType_KEY_TYPE_ECDSA_SECP256K1))
 		require.Equal(t, newKeyRequestEvents[0].Creator, alice.EthAddress(t))
 
 		actions3, err := iActClient.Actions(alice.CallOps(t), act.TypesPageRequest{})
 
 		require.NoError(t, err)
 		require.Len(t, actions3.Actions, 3)
 
-		keyRequests1, err := iWardenClient.KeyRequests(alice.CallOps(t), warden.TypesPageRequest{}, 1, int32(types.KeyRequestStatus_KEY_REQUEST_STATUS_PENDING), 1)
+		keyRequests1, err := iWardenClient.KeyRequests(alice.CallOps(t), warden.TypesPageRequest{}, 1, uint8(types.KeyRequestStatus_KEY_REQUEST_STATUS_PENDING), 1)
```
[the remark this answers is not in the document]
Comment (Svetomech) [reply tests/cases/warden_precompile_action.go:186]: outside of scope of this PR

File: tests/cases/warden_precompile.go [hunk 3]
```diff
@@ -294,7 +294,7 @@ func (c *Test_WardenPrecompile) Run(t *testing.T, ctx context.Context, build fra
 
 		require.Len(t, newKeyEvents, 1)
 		require.Equal(t, uint64(1), newKeyEvents[0].Id)
-		require.Equal(t, int32(1), newKeyEvents[0].KeyType)
+		require.Equal(t, uint8(1), newKeyEvents[0].KeyType)
```
[the remark this answers is not in the document]
Comment (Svetomech) [reply tests/cases/warden_precompile.go:297]: This is expected behaviour

File: precompiles/warden/query_types.go [hunk 4]
```diff
@@ -578,8 +578,8 @@ func (kr *KeyRequest) mapKeyRequest(keyRequest types.KeyRequest) (*KeyRequest, e
 	kr.Creator = ethCreator
 	kr.SpaceId = keyRequest.SpaceId
 	kr.KeychainId = keyRequest.KeychainId
-	kr.KeyType = int32(keyRequest.KeyType)
-	kr.Status = int32(keyRequest.Status)
+	kr.KeyType = uint8(keyRequest.KeyType)
+	kr.Status = uint8(keyRequest.Status)
```
[the remark this answers is not in the document]
Comment (Svetomech) [reply precompiles/warden/query_types.go:582]: @mn13 what do you think about this validation, did we miss it before?

File: precompiles/justfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (Svetomech) [reply precompiles/justfile]: done

File: precompiles/justfile [hunk 5]
```diff
@@ -4,15 +4,26 @@
...
+  abigen --abi {{git_root}}/precompiles/warden/abi.json --pkg warden --type IWarden --out {{git_root}}/precompiles/warden/IWarden.go
+  cat {{tmpdir}}/IWarden.abi | jq '. | {abi: .}' > {{git_root}}/precompiles/warden/abi.json
 
-  cat {{git_root / "precompiles/slinky/abi.json"}} | jq '.abi' > {{tmpdir / "slinky.json"}}
-  abigen --abi {{tmpdir / "slinky.json"}}  --pkg slinky --type ISlinky --out {{git_root / "precompiles/slinky/ISlinky.go"}}
+  # Generate Slinky contract artifacts
+  solc --evm-version paris --abi {{git_root}}/precompiles/slinky/ISlinky.sol -o {{tmpdir}} --overwrite
+  cat {{tmpdir}}/ISlinky.abi | jq '.' > {{git_root}}/precompiles/slinky/abi.json
+  abigen --abi {{git_root}}/precompiles/slinky/abi.json --pkg slinky --type ISlinky --out {{git_root}}/precompiles/slinky/ISlinky.go
+  cat {{tmpdir}}/ISlinky.abi | jq '. | {abi: .}' > {{git_root}}/precompiles/slinky/abi.json
```
[the remark this answers is not in the document]
Comment (Svetomech) [reply precompiles/justfile]: done

File: precompiles/warden/query_types.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (mn13) [reply precompiles/warden/query_types.go:582]: No, it's no make sense it this context, as I know this processed by proto

**Review by mn13 — APPROVED** [review]
LGTM!

**Review by pitasi — APPROVED** [review]

**Pull request merged**